diff --git a/screencast/pipewirestream.cpp b/screencast/pipewirestream.cpp
index ff813b7c4..91dfff0cd 100644
--- a/screencast/pipewirestream.cpp
+++ b/screencast/pipewirestream.cpp
@@ -244,7 +244,9 @@ bool PipeWireStream::createStream()
                                         SPA_TYPE_OBJECT_Format, SPA_PARAM_EnumFormat,
                                         SPA_FORMAT_mediaType, SPA_POD_Id(SPA_MEDIA_TYPE_video),
                                         SPA_FORMAT_mediaSubtype, SPA_POD_Id(SPA_MEDIA_SUBTYPE_raw),
-                                        SPA_FORMAT_VIDEO_format, SPA_POD_Id(format),
+                                        SPA_FORMAT_VIDEO_format, SPA_POD_CHOICE_ENUM_Id(format == SPA_VIDEO_FORMAT_BGRA ? 3 : 2,
+                                                                                        format, format, format == SPA_VIDEO_FORMAT_BGRA ? SPA_VIDEO_FORMAT_BGRx : SPA_VIDEO_FORMAT_UNKNOWN),
+
                                         SPA_FORMAT_VIDEO_size, SPA_POD_Rectangle(&resolution),
                                         SPA_FORMAT_VIDEO_framerate, SPA_POD_Fraction(&defaultFramerate),
                                         SPA_FORMAT_VIDEO_maxFramerate, SPA_POD_CHOICE_RANGE_Fraction(&maxFramerate, &minFramerate, &maxFramerate));
